cfme/tests/cloud/test_cloud_timelines.py

```diff
@@ -1,164 +1,92 @@
-    ems_events = db[ems_events_table_name]
-    with db.transaction:
-        providers = (
-            db.session.query(ems_events.id)
-            .join(ems, ems_events.ems_id == ems.id)
-            .filter(ems.name == provider.name)
-        )
-        db.session.query(ems_events).filter(ems_events.id.in_(providers.subquery())).delete(False)
+def test_instance(request, cloud_provider):
+    instance_name = "test_tl_" + fauxfactory.gen_alphanumeric(length=4)
+    instance = Instance.factory(instance_name, cloud_provider)
 
+    request.addfinalizer(instance.delete_from_provider)
 
-@pytest.fixture(scope="module")
-def vm_name():
-    # We have to use "tt" here to avoid name truncating in the timelines view
-    return "test_tt_{}".format(fauxfactory.gen_alphanumeric())
+    if not cloud_provider.mgmt.does_vm_exist(instance.name):
+        logger.info("deploying %s on provider %s", instance.name, cloud_provider.key)
+        instance.create_on_provider(allow_skip="default", find_in_cfme=True)
+    return instance
 
 
 @pytest.fixture(scope="module")
-def delete_instances_fin(request):
-    """ Fixture to add a finalizer to delete provisioned instances at the end of tests
+def gen_events(test_instance):
+    logger.debug('Starting, stopping VM')
+    mgmt = test_instance.provider.mgmt
+    mgmt.stop_vm(test_instance.name)
+    mgmt.start_vm(test_instance.name)
 
-    This is a "trashbin" fixture - it returns a mutable that you put stuff into.
-    """
-    provisioned_instances = {}
 
-    def delete_instances(instances_dict):
-        for instance in instances_dict.itervalues():
-            instance.delete_from_provider()
-    request.addfinalizer(lambda: delete_instances(provisioned_instances))
-    return provisioned_instances
+def count_events(target, vm):
+    timelines_view = navigate_to(target, 'Timelines')
+    timelines_view.filter.time_position.select_by_visible_text('centered')
+    timelines_view.filter.apply.click()
+    found_events = []
+    for evt in timelines_view.chart.get_events():
+        # BZ(1428797)
+        if not hasattr(evt, 'source_vm'):
+            logger.warn("event {evt!r} doesn't have source_vm field. "
+                        "Probably issue".format(evt=evt))
+            continue
+        elif evt.source_vm == vm.name:
+            found_events.append(evt)
 
-
-@pytest.mark.uncollectif(lambda: version.current_version() >= '5.7')
-@pytest.fixture(scope="module")
-def test_instance(setup_provider_modscope, request, delete_instances_fin, provider, vm_name):
-    """ Fixture to provision instance on the provider
-    """
-    instance = Instance.factory(vm_name, provider)
-    if not provider.mgmt.does_vm_exist(vm_name):
-        delete_instances_fin[provider.key] = instance
-        instance.create_on_provider(allow_skip="default")
-    return instance
+    logger.info("found events: {evt}".format(evt="\n".join([repr(e) for e in found_events])))
+    return len(found_events)
```

Comment:
exactly, my current goal  is to fix existing tests. Old tests checked only that some events were received.
 I'll completely refactor those tests soon.